### PR TITLE
Fix dtb warning reporting

### PR DESCRIPTION
**Pull request by ConchuOD**
Similarly to the rv64_build_warn test, drop the extra build from the dtb test.
Fix the actually reporting too - it cannot use the same checking as the rv64_build_warn test that I copied it from.

### CODE REVIEW

File: tests_rv/patch/dtb_warn_rv64/dtb_warn_rv64.sh [hunk 1]
```diff
@@ -38,42 +31,32 @@ make -C . O=$tmpdir0 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
...
-    > $tmpfile_fn
-
-  diff -U 0 $tmpfile_fo $tmpfile_fn 1>&2
-  rm $tmpfile_fo $tmpfile_fn
-
   rc=1
 fi
 
-rm -rf $tmpdir0 $tmpfile_o $tmpfile_n
+#rm -rf $tmpdir0 $tmpfile_o $tmpfile_n
```
Comment (bjoto) [inline tests_rv/patch/dtb_warn_rv64/dtb_warn_rv64.sh]: Leaking tempfiles and the new tmpdir1 and tmpdir0? (Are both used?)

Comment (ConchuOD) [reply tests_rv/patch/dtb_warn_rv64/dtb_warn_rv64.sh]: Woops, I was keeping them locally when I was testing. Also, it's easier to build in two directories than do a clean. I'll add in the second dir and uncomment it.
Thanks.

Comment (ConchuOD) [reply tests_rv/patch/dtb_warn_rv64/dtb_warn_rv64.sh]: > Woops, I was keeping them locally when I was testing. Also, it's easier to build in two directories than do a clean. I'll add in the second dir and uncomment it. Thanks.

Done, & renamed the directories to match their usage.

**Review by bjoto — APPROVED** [review]
LGTM!

**Pull request merged**